src/escrows/DolaFraxBPEscrow.sol

```diff
@@ -0,0 +1,271 @@
+    */
+    function disallowClaimOnBehalf(address allowee) external onlyBeneficiary {
+        allowlist[allowee] = false;
+        emit AllowClaim(allowee, false);
+    }
+
+    /**
+     * @notice Deposit token balance in the escrow into Convex
+     * @dev Cannot deposit if there are Yearn tokens in the escrow (only 1 strategy at a time)
+     */
+    function depositToConvex() external onlyBeneficiary {
+        if (yearn.balanceOf(address(this)) > 0)
+            revert CannotDepositToConvex(yearn.balanceOf(address(this)));
+        uint256 tokenBal = token.balanceOf(address(this));
+        stakedBalance += tokenBal;
+        booster.deposit(pid, tokenBal, true);
+    }
+
+    /**
+     * @notice Deposit token balance in the escrow into Yearn
+     * @dev Cannot deposit if there are Convex tokens in the escrow (only 1 strategy at a time)
+     */
+    function depositToYearn() external onlyBeneficiary {
+        if (stakedBalance > 0) revert CannotDepositToYearn(stakedBalance);
+        yearn.deposit(token.balanceOf(address(this)), address(this));
+    }
+
+    /**
+     * @notice Withdraw all tokens from Yearn
+     * @return lpAmount The amount of tokens withdrawn from Yearn
+     */
+    function withdrawFromYearn()
+        external
+        onlyBeneficiary
+        returns (uint256 lpAmount)
+    {
+        return _fullWithdrawFromYearn();
+    }
+
+    /**
+     * @notice Withdraw all tokens from Convex
+     * @return lpAmount The amount of tokens withdrawn from Convex
+     */
+    function withdrawFromConvex()
+        external
+        onlyBeneficiary
+        returns (uint256 lpAmount)
+    {
+        return _fullWithdrawFromConvex();
+    }
+
+    /**
+     * @notice Move all tokens deposited into Convex to Yearn
+     * @dev Will move all tokens from Convex to Yearn, including extra lp tokens that might be in the escrow if useAll is true
+     * @param useAll If true, deposit the full balance in the escrow
+     * @return lpAmount The amount of tokens deposited into Yearn
+     */
+    function moveFromConvexToYearn(
+        bool useAll
+    ) external onlyBeneficiary returns (uint256 lpAmount) {
+        lpAmount = _fullWithdrawFromConvex();
+
+        if (useAll) lpAmount = token.balanceOf(address(this));
+
+        yearn.deposit(lpAmount, address(this));
+    }
+
+    /**
+     * @notice Move all tokens deposited into Yearn to Convex
+     * @dev Will move all tokens from Yearn to Convex, including extra lp tokens that might be in the escrow if useAll is true
+     * @param useAll If true, deposit the full balance in the escrow
+     * @return lpAmount The amount of tokens deposited into Convex
+     */
+    function moveFromYearnToConvex(
+        bool useAll
+    ) external onlyBeneficiary returns (uint256 lpAmount) {
+        lpAmount = _fullWithdrawFromYearn();
+
+        if (useAll) lpAmount = token.balanceOf(address(this));
+
+        stakedBalance += lpAmount;
+        booster.deposit(pid, lpAmount, true);
+    }
```

Comment:
Unnecessary if implementing automatic withdraws in deposit functions

Reply:
Imo having the user deposit all into the strategy they decide on is the way to go, but worth bringing up with others if they think it's better to let the user choose.